lib/Doctrine/ODM/PHPCR/Mapping/ClassMetadataFactory.php

```diff
@@ -155,7 +159,6 @@ private function addInheritedDocumentOptions(ClassMetadata $subClass, ClassMetad
         $subClass->setVersioned($parentClass->versionable);
         $subClass->setReferenceable($parentClass->referenceable);
         $subClass->setNodeType($parentClass->getNodeType());
-        $subClass->setMixins($parentClass->getMixins());
```

Comment:
@lsmith77 do you see the reason why this did not always let the mixins of the parent class win and prevent ever overwriting the mixins? same actually for the other options - we do not check if something is already set...